codecs/mozjpeg_enc/mozjpeg_enc.cpp

```diff
@@ -109,18 +118,48 @@ void encode(uint8_t* image_buffer, int image_width, int image_height, int qualit
    */
   cinfo.image_width = image_width;      /* image width and height, in pixels */
   cinfo.image_height = image_height;
-  cinfo.input_components = 3;           /* # of color components per pixel */
-  cinfo.in_color_space = JCS_RGB;       /* colorspace of input image */
+  cinfo.input_components = 4;           /* # of color components per pixel */
+  cinfo.in_color_space = JCS_EXT_RGBA;  /* colorspace of input image */
   /* Now use the library's routine to set default compression parameters.
    * (You must set at least cinfo.in_color_space before calling this,
    * since the defaults depend on the source color space.)
    */
   jpeg_set_defaults(&cinfo);
+
   /* Now you can set any non-default parameters you wish to.
    * Here we just illustrate the use of quality (quantization table) scaling:
    */
-  jpeg_set_quality(&cinfo, quality, TRUE /* limit to baseline-JPEG values */);
+  jpeg_set_colorspace(&cinfo, (J_COLOR_SPACE) opts.color_space);
+
+  if (opts.quant_table != -1) {
+    jpeg_c_set_int_param(&cinfo, JINT_BASE_QUANT_TBL_IDX, opts.quant_table);
+  }
 
+  cinfo.optimize_coding = opts.optimize_coding ? TRUE : FALSE;
+
+  if (opts.arithmetic) {
+    cinfo.arith_code = TRUE;
+    cinfo.optimize_coding = FALSE;
+  }
+
+  cinfo.smoothing_factor = opts.smoothing;
+
+  jpeg_c_set_bool_param(&cinfo, JBOOLEAN_USE_SCANS_IN_TRELLIS, opts.trellis_multipass ? TRUE : FALSE);
```

Comment:
Are the ternaries necessary here?

Reply:
Good question. I assumed not since C doesn't have a real boolean type right? I'll test anyway.

Reply:
I mean, you are using `bool` in the struct definition, so I’d assume the compiler does the right thing ™️  here

Reply:
Yes it does.